daft_dashboard/daft_dashboard/__init__.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+import os
+from pathlib import Path
+from daft_dashboard import daft_dashboard as native
+from importlib import resources
+
+
+DAFT_DASHBOARD_ENV_NAME = "DAFT_DASHBOARD"
+DAFT_DASHBOARD_URL = "http://localhost:3238/api/queries"
+
+
+def launch(block: bool = False):
```

Comment:
this should fail if there is already something running on that port. 

```py
from daft import dashboard
dashboard.launch() # ok
dashboard.launch() # fails as something is already running on port 3238
```

Reply:
Actually, I can throw an exception here and put the onus on the end-user to `try-except` that launch. That way, they can achieve idempotency if they want to.

Reply:
`launch` and `shutdown` both throw an exception if a Daft dashboard server process has already been launched/shutdown, respectively.

Both of them also have a `noop_if_initialized`/`noop_if_shutdown` flag, respectively, to disable the exception throwing.